manifests/certonly.pp

```diff
@@ -56,62 +67,97 @@
   Variant[Integer[0,23], String, Array]     $cron_hour            = fqdn_rand(24, $title),
   Variant[Integer[0,59], String, Array]     $cron_minute          = fqdn_rand(60, fqdn_rand_string(10, $title)),
   Stdlib::Unixpath                          $config_dir           = $letsencrypt::config_dir,
+  Array[String]                             $pre_hook_commands    = [],
```

Comment:
please also enforce the minimal string length:
```suggestion
  Array[String[1]]                          $pre_hook_commands    = [],
```

Reply:
Ah, wait .. won't this fail on empty arrays? Or does it still validate with just `[]`?

Reply:
Nevermind, I see it now: Minimal string length of 0 or more elements obviously has nothing to do with minimal array length.